internal/datacoord/meta.go

```diff
@@ -1274,6 +1248,25 @@ func (m *meta) GetEarliestStartPositionOfGrowingSegments(label *CompactionGroupL
 	return earliest
 }
 
+func (m *meta) GetHasUnindexTaskSegments() []*SegmentInfo {
```

Comment:
Would it be better to refactor this into a utility function? The implementation of utility function could be: 
```
func (s *Server)GetHasUnindexTaskSegments() []*SegmentInfo{
    flushedSegments := s.meta.SelectSegment(func(seg *SegmentInfo) bool {
         return isFlush(seg)
    })

    for _, segID := range flushedSegments {
        s.indexMeta.IsUnindexSegment(seg)
        ....
    }
}
```

This way, we can avoid having two layers of locks.